test_runner/src/main/kotlin/ftl/filter/TestFilters.kt

```diff
@@ -150,12 +150,20 @@ object TestFilters {
         }
     )
 
-    private fun withClassName(classNames: List<String>): TestFilter = TestFilter(
-        describe = "withClassName (${classNames.joinToString(", ")})",
-        shouldRun = { testMethod ->
-            withPackageName(classNames).shouldRun(testMethod)
-        }
-    )
+    private fun withClassName(classNames: List<String>): TestFilter {
+        val splittedClassNames = classNames.map { it.split("#") }
+        return TestFilter(
+            describe = "withClassName (${classNames.joinToString(", ")})",
+            shouldRun = { testMethod ->
+                val splittedName = testMethod.testName.split("#")
+                splittedClassNames.any { className ->
+                    // className.size == 1 => foo.bar.TestClass1
+                    // className.size != 1 => foo.bar.TestClass1#testMethod1
+                    splittedName[0] == className[0] && (className.size == 1 || splittedName[1] == className[1])
```

Comment:
I know that there is a comment, but could you wrap this code in some function(s) with good naming, because next month I think that it will be hard to understand

Reply:
Good point, I'll rewrite it to be more self-explanatory

Reply:
@pawelpasterz  how about making it:
```kotlin

    private fun withClassName(classNames: List<String>): TestFilter {
        // splits foo.bar.TestClass1#testMethod1 into [foo.bar.TestClass1, testMethod1]
        fun String.extractClassAndTestNames() = split("#")
        val classFilters = classNames.map { it.extractClassAndTestNames() }
        return TestFilter(
            describe = "withClassName (${classNames.joinToString(", ")})",
            shouldRun = { testMethod -> testMethod.testName.extractClassAndTestNames().matchFilters(classFilters) }
        )
    }
    
    private fun List<String>.matchFilters(classFilters: List<List<String>>): Boolean {
        fun List<String>.className() = first()
        fun List<String>.methodName() = last()
        return classFilters.any { filter ->
            // When filter.size == 1 all test methods from the class should run therefore we do not compare method names
            // When filter.size != 1 only particular test from the class should be launched and we need to compare method names as well
            className() == filter.className() && (filter.size == 1 || methodName() == filter.methodName())
        }
    }
```